Fix title of site label in the filter navigator

tirtharajghosh opened the pull request:
I have fixed the title. Now it will show the full url with protocol (eg. `https://example.com`) instead of [Object object].  Fixes #2791

canova reviewed (changes requested):
Hi @tirtharajghosh! Thanks a lot for the PR, looks pretty good to me!

I left some minor comments about naming and making small changes. But there is one blocker we need to fix before landing this. Currently these snapshot changes indicate that we are breaking something with the other part of the code. 

Steps to reproduce: 
1. Open the profile mentioned in the issue:
2. Select a portion of the timeline by clicking and dragging an area of it. 
3. Click on the magnifier button to zoom in the timeline.

If you repeat these steps a few times, you'll see something like this in the picture.
<img width="381" alt="Screen Shot 2020-09-29 at 2 46 03 PM" src="https://user-images.githubusercontent.com/466239/94561551-75b95f00-0264-11eb-8c37-2594cdd75bf6.png">

But it should show something like this instead:
 
<img width="475" alt="Screen Shot 2020-09-29 at 3 01 59 PM" src="https://user-images.githubusercontent.com/466239/94561785-c7fa8000-0264-11eb-8092-6ddbe2c6122c.png">


This is happening becasue, we are changing the `items` field of `FilterNavigatorBar` from a `React.Node` array to an `object` array. But we are still sending `string` arrays for places that use the return values of `getCommittedRangeLabels` and `getTransformLabels`. So we should still take it into account that we can get string values. 

To fix this, I would suggest making `FilterNavigatorBar` component accept both that object and string as `items` value. So it should look something like this:
```js
+items: $ReadOnlyArray<
    | {
        title?: string,
        data: React.Node,
      }
    | string
  >
```

Then we need to also fix the places where we use `items` object.
For example:
`title={item.title}` should be `{typeof item === 'string' ? item : item.title}` instead to both work with strings and objects.

Thanks again for working on this! Let me know if you need to ask any question here or in the Matrix channel on https://chat.mozilla.org/#/room/#profiler:mozilla.org .

tirtharajghosh commented:
Hi @canova ! Thanks for the detailed explanations of the errors. I have fixed all the issues and changed all the non-specific naming as you requested.
Please let me know if i missed anything or anything is incorrect.

canova reviewed (approved):
Thanks for working on this @tirtharajghosh, looks pretty good to me!
